pages/news/2020-05-16-intro-outro-sections.html

```diff
@@ -0,0 +1,107 @@
+title: Intro & Outro Cues, Silence Detection, and AutoDJ Improvements
+author: Be.
+date: 2020-05-16 01:00:00
+
+{% extends "post.html" %}
+
+{% block post %}
+
+{% load markup %}
+{% filter markdown %}
+
+Do you use hotcues to mark the points you use to mix in and out of tracks? Have you wished AutoDJ knew to mix in and out at those points?
+In Mixxx 2.3, we have introduced new intro & outro cues for these purposes.
+These are not ordinary cue points. Unlike hotcues, and unlike intro and outro markers in other DJ software, they do not mark only one point; the intro & outro are sections and each is defined by two points.
+
+When analyzing a track, Mixxx detects where the first and last sounds are to make it easy to skip playing silent parts at the beginning and end of the track.
+The analyzer places the intro start point at the first sound and the outro end point at the last sound.
+<!-- screenshot -->
+
+It is up to you to decide where to place the intro end and outro start points.
+You can easily jump to the intro end and outro start points by pressing the buttons like hotcues, but that is not their primary purpose.
+Marking the whole intro and outro allows Mixxx to calculate how long each section is and show it on the overview waveform.
+You can use this information to inform when you press play on the next track and so can AutoDJ.
+<!-- screenshot -->
+
+## Intro & Outro Mixing Techniques Before Mixxx 2.3 ##
+
+Before we implemented these intro & outro cues, I calculated how long these sections were in my head.
+After I identified which track I wanted to mix in next, I would find the point in that track where I would want to have the old track completely faded out.
+Then, I would look at what time that was on the deck and keep that number in my memory.
+I would watch the how much time was remaining on the old track and listen for the time to start mixing out the old track.
+
+For example, if I knew the old track had a fairly sudden ending with only 1 or 2 seconds without a lot going on at the end, and the next track had a 20 second intro, to avoid letting the energy of the mix drop if I let only that intro play, I could start the next track when the old track's deck showed 20 seconds remaining.
+This would align the end of the old track with the end of the intro of the new track.
```

Comment:
```suggestion
This would align the end of the old track with the end of the intro of the new track so that the energy of the new track would pick up right as the old track ends.
```